src/lib/PPrint.hs

```diff
@@ -361,12 +361,18 @@ instance Pretty (PiType r n) where
     in prettyBinder <> (group $ line <> p arr <> prettyEff <> prettyBody)
 
 instance Pretty (TabPiType r n) where
-  pretty (TabPiType (b :> IxType ty _) body) = let
+  pretty (TabPiType (b :> IxType ty dict) body) = let
     prettyBinder = prettyBinderHelper (b:>ty) body
     prettyBody = case body of
       Pi subpi -> pretty subpi
       _ -> pLowest body
-    in prettyBinder <> (group $ line <> "=>" <+> prettyBody)
+    in prettyBinder <> prettyIxDict dict <> (group $ line <> "=>" <+> prettyBody)
+
+-- A helper to let us turn dict printing on and off.  We mostly want it off to
+-- reduce clutter in prints and error messages, but when debugging synthesis we
+-- want it on.
+prettyIxDict :: Atom r n -> Doc ann
+prettyIxDict dict = if False then " " <> p dict else mempty
```

Comment:
You could enable it based on an env var. We do that same trick in a few other places I think